core/common/src/main/java/alluxio/worker/block/BlockWorker.java

```diff
@@ -115,6 +115,28 @@ void commitBlock(long sessionId, long blockId, boolean pinOnCreate)
   String createBlock(long sessionId, long blockId, int tier, String medium, long initialBytes)
       throws BlockAlreadyExistsException, WorkerOutOfSpaceException, IOException;
 
+  /**
+   * Creates a block in Alluxio managed space.
+   * Calls {@link #createBlockWriter} to get a writer for writing to the block.
+   * The block will be temporary until it is committed by {@link #commitBlock} .
+   * Throws an {@link IllegalArgumentException} if the location does not belong to tiered storage.
+   *
+   * @param sessionId the id of the client
+   * @param blockId the id of the block to create
+   * @param tier the tier to place the new block in
+   *        {@link BlockStoreLocation#ANY_TIER} for any tier
+   * @param alluxioPath the alluxio path
+   * @param medium the name of the medium to place the new block in
+   * @param initialBytes the initial amount of bytes to be allocated
+   * @return a string representing the path to the local file
+   * @throws BlockAlreadyExistsException if blockId already exists, either temporary or committed,
+   *         or block in eviction plan already exists
+   * @throws WorkerOutOfSpaceException if this Store has no more space than the initialBlockSize
+   */
+  String createBlock(long sessionId, long blockId, int tier, String alluxioPath,
```

Comment:
Possibly to have an `CreateBlockOptions` rather than adding a new API method with a new arg?

Something like 

`String createBlock(long sessionId, long blockId, int tier, CreateBlockOptions options)`
and in this `CreateBlockOptions` there are fields like medium, initialBytes and alluxioPath.
These three are not required fields, thus we put them in options